src/appleseed.studio/mainwindow/renderingsettingswindow.cpp

```diff
@@ -790,13 +790,27 @@ namespace
             const bool              allow_unlimited = true)
         {
             const int DefaultMaxBounces = 8;
+            const int DefaultMaxDiffuseBounces = 3;
 
             const int max_bounces =
                 get_config<int>(config, construct_bounce_param_path(bounce_type), -1);
-
+	    
+            const string widget_max_bounce_key = widget_key_prefix + ".bounces.max_" + bounce_type + "_bounces";
+            const string widget_unlimited_max_bounce_key = widget_key_prefix + ".bounces.unlimited_" + bounce_type + "_bounces";
+            
             if (allow_unlimited)
-                set_widget(widget_key_prefix + ".bounces.unlimited_" + bounce_type + "_bounces", max_bounces == -1);
-            set_widget(widget_key_prefix + ".bounces.max_" + bounce_type + "_bounces", max_bounces == -1 ? DefaultMaxBounces : max_bounces);
+            {   
+                if (bounce_type == "diffuse")
```

Comment:
I think we misunderstood each others. The logic should not change, it was right before (before your PR, and in your initial commit).